sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/NullPointerDereferenceCSharp6.cs

```diff
@@ -58,5 +59,106 @@ public void TryCatch4(object o)
                 var b = o.ToString(); // Noncompliant, o could be null here
             }
         }
+
+    public void Compliant(List<int> list)
+    {
+      var row = list?.Count;
+      if (row != null)
+      {
+        var type = list.ToArray();
+      }
+    }
+
+    public class A
+    {
+      public bool booleanVal { get; set; }
+    }
+
+    public void Compliant1(List<int> list, A a)
+    {
+      var row = list?.Count;
+      if (a.booleanVal = (row != null))
+      {
+        var type = list.ToArray();
+      }
+    }
+
+    public void NonCompliant(List<int> list)
+    {
+      var row = list?.Count;
+      if (row == null)
+      {
+        var type = list.ToArray(); // Noncompliant
+      }
+    }
+
+    public void NonCompliant1(List<int> list, A a)
+    {
+      var row = list?.Count;
+      if (a.booleanVal = (row == null))
+      {
+        var type = list.ToArray(); // Noncompliant
+      }
     }
+
+    void Compliant2(object o)
+    {
+      switch (o?.GetHashCode())
+      {
+        case 1:
+          o.ToString();
+          break;
+        default:
+          break;
+      }
+    }
+
+    void NonCompliant2()
+    {
+      object o = null;
+      switch (o?.GetHashCode())
+      {
+        case null:
+          o.ToString(); // Noncompliant
+          break;
+        default:
+          break;
+      }
+    }
+  }
+
+  public class S2259
```

Comment:
I'd replace `S2259` with `ReproForIssue2338` or some random name.

all tests in this file are for `S2259`